consensus/src/chained_bft/persistent_storage.rs

```diff
@@ -170,18 +217,18 @@ impl<T: Payload> RecoveryData<T> {
     /// and the ledger info for the root block, return an error if it can not be found.
     ///
     /// We guarantee that the block corresponding to the storage's latest ledger info always exists.
-    /// In the case of an epoch boundary ledger info(i.e. it has validator set), we generate the virtual genesis block.
     fn find_root(
         blocks: &mut Vec<Block<T>>,
         quorum_certs: &mut Vec<QuorumCert>,
         storage_ledger: &LedgerInfo,
-    ) -> Result<(Block<T>, QuorumCert, QuorumCert)> {
+    ) -> Result<RootInfo<T>> {
         let root_from_storage = storage_ledger.consensus_block_id();
         info!(
             "The last committed block id as recorded in storage: {}",
             root_from_storage
         );
 
+        // TODO: remove genesis building logic here, this should be done in event processor wrapper
```

Comment:
sorry I forget what's the motivation of moving this?

Reply:
Since `genesis_block` is constructed in `LedgerRecoveryData` now, I don't think we need to construct it here again. 
Please correct me if I'm wrong, but what currently `find_root` is doing with `genesis_block` case is that it constructs the block, pushes the block and qc into `blocks` and `quorum_certs `, then it founds out the index of the `genesis_block` in the `blocks` and pop the block out of it. Then it finds the genesis qc in the `quorum_certs` and clone it twice...The thing `find_root` is doing is suitable for a non-genesis root, but if the root is a genesis root I think the process can be simplified. We don't need to `find_root` in genesis case because we already know everything about the root.